lib/chore/job.rb

```diff
@@ -108,6 +114,25 @@ def prefixed_queue_name
       def has_backoff?
         self.options.key?(:backoff)
       end
+
+      def has_dedupe_lambda?
+        self.options.key?(:dedupe_lambda)
+      end
+
+      def dedupe_key(*args)
+        return unless has_dedupe_lambda?
+        if self.options[:dedupe_lambda].arity != self.instance_method(:perform).arity
+          raise ArgumentError, "#{self.to_s}: dedupe_lambda must accept the same number of arguments as perform"
+        end
```

Comment:
At this point, I feel like this type of error checking is unnecessary.  You'll effectively get the same message when you execute `call` with a different set of arguments.  If someone feels like using a `Proc`, then they'll be able to do that.